### DISCUSSION TITLE
404 Error Accessing /keystatic Route After Enabling Astro's Built-in i18n Feature

### DISCUSSION THREAD

**Discussion by lefpap** [Thinkmill/keystatic#1057] [Q&A] [answered]
## The problem
Hello, I'm currently working on a project using Astro and integrating keystatic for content management. Initially, everything worked fine, and I was able to access the keystatic admin panel through the `/keystatic` route without any issues.

However, after enabling Astro's built-in internationalization (i18n) feature, which required changing my routing to include a top-level directory for each locale (e.g., [locale].astro), I've encountered a problem. Now, when I try to access the /keystatic route, I receive a 404 Page Not Found error.

My pages directory looks like this:
```
/pages
   |_ [locale].astro
        |_ index.astro
   |_ index.astro
```

Basicaly the top level `index.astro` is there because it needs to exist in there and has no content inside.

## Steps to Reproduce:

1. Set up a basic Astro project with keystatic integration.
2. Enable Astro's i18n feature.
3. Create a top level route for the locales (e.g. [locale].astro) and create a page using your locales and the `getStaticPaths` method.
4. Attemp to access any of `/keystatic`, `/en/keystatic` or `/keystatic/en/` routes.

## Expected Behavior:
Access to the /keystatic route should remain unaffected by the addition of locale-specific routing, allowing me to manage my content as before.

## Actual Behavior:
After enabling i18n and creating a [locale].astro top level route, accessing the /keystatic route results in a 404 Page Not Found error.

## Question:
Is there a known workaround or configuration change that needs to be applied to maintain access to the `/keystatic` route while utilizing Astro's i18n feature and this structure for the routing? Any guidance or suggestions would be greatly appreciated.

Thank you in advance for your help!

**Comment by stefanprobst** [comment]
- using the `prefixDefaultLocale: false` astro i18n routing strategy should work
- when using `prefixDefaultLocale: true`, there are currently a couple of issues. in the upcoming astro 4.6 it should be able to work around these with a "manual" i18n routing strategy - see [the pr](https://github.com/withastro/astro/pull/10193) for this

**Comment by lefpap** [comment, marked as answer]
Yeah I did this and I had to change the `/[locale]` directory to `/[...locale]` because single routes can't have `undefined` as value. I don't know if this is the right way.

**Comment by simonswiss** [comment]
Collating this in https://github.com/Thinkmill/keystatic/issues/1080 — there are numerous requests and issues around supporting multi-lingual sites in Keystatic.
